### PR TITLE
[CUR-439] Refactor units tab data 

### PR DESCRIPTION
**Pull request by sonalisin**
## Description

Music year: 2016

- Moved useEffect in UnitsTab to [tab].tsx 
- Removed 'data' and 'examboardSlug' as props from UnitsTab
- Added trackingData prop to units tab 
- Added temporary unitTab fix for weird state changes to yearSelection in units tab (useLayoutEffect) until url routing is added to visualiser
- Updated tests & tidied up tests
Note: Passed duplicateSlugs from data formatting as a Set -> array -> Set as getStaticProps didn't like non-serializable types

## How to test

1. Go to https://deploy-preview-2411--oak-web-application.netlify.thenational.academy
2. Click on the curriculum visualiser
3. You should see no changes from the current state of the curriculum visualiser (🤞)

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/components/CurriculumComponents/CurriculumVisualiser/CurriculumVisualiser.tsx [hunk 1]
```diff
@@ -351,14 +352,15 @@ const CurriculumVisualiser: FC<CurriculumVisualiserProps> = ({
                   data-testid="unit-cards"
                 >
                   {units
-                    .filter((unit: Unit) =>
-                      isVisibleUnit(
+                    .filter((unit: Unit) => {
+                      const isVisible = isVisibleUnit(
```
Comment (orangemug) [inline src/components/CurriculumComponents/CurriculumVisualiser/CurriculumVisualiser.tsx:356]: **nitpick**: no need to change this now, but it's good practice to not have non-functional changes in PRs, as it makes git history harder to read

Really not a big issue :)

File: src/components/CurriculumComponents/UnitsTab/UnitsTab.test.tsx [hunk 2]
```diff
@@ -486,26 +583,39 @@ describe("components/pages/CurriculumInfo/tabs/UnitsTab", () => {
...
+        formattedData={formatCurriculumUnitsData(curriculumUnitsTabFixture())}
+      />,
     );
 
     const yearOptions = (await queryAllByTestId(
       "year-radio",
     )) as HTMLInputElement[];
 
-    const year10Option = yearOptions.find((option) => option.value === "10");
+    const year10Option = yearOptions.find((option) => option.value === "7");
```
Comment (orangemug) [inline src/components/CurriculumComponents/UnitsTab/UnitsTab.test.tsx:606]: Variable still says `year10...` but we're finding year 7

Comment (orangemug) [reply src/components/CurriculumComponents/UnitsTab/UnitsTab.test.tsx:606]: Also more reference to `year10` below this line

File: src/pages/teachers/curriculum/[subjectPhaseSlug]/[tab].tsx [hunk 3]
```diff
@@ -171,6 +224,179 @@ export const parseSubjectPhaseSlug = (slug: string) => {
...
+): CurriculumUnitsFormattedData {
+  const threadOptions = [] as Thread[];
+  const yearOptions = [] as string[];
+  const yearData = {} as CurriculumUnitsYearData;
+  const initialYearSelection = {} as YearSelection;
+
+  const unitSlugs = new Set<string>();
+  const duplicateUnitSlugs = new Set<string>();
+
+  unitSlugs.clear();
```
Comment (orangemug) [inline src/pages/teachers/curriculum/[subjectPhaseSlug]/[tab].tsx:238]: I don't think these lines do anything as these `Set`s are created a few lines above

File: src/pages/teachers/curriculum/[subjectPhaseSlug]/[tab].tsx [hunk 4]
```diff
@@ -171,6 +224,179 @@ export const parseSubjectPhaseSlug = (slug: string) => {
   };
 };
 
+export function formatCurriculumUnitsData(
```
Comment (orangemug) [inline src/pages/teachers/curriculum/[subjectPhaseSlug]/[tab].tsx:227]: I think it'd be good to test `formatCurriculumUnitsData(...)` standalone with lots of different input data (battle test) as it seems quite important that we've got this part correct.

**Review by rightsaidjames — APPROVED** [review]
Briefly retested following additional changes - all looks good!

**Review by orangemug — APPROVED** [review]

**Pull request merged**